app/models/work_package/pdf_export/markdown.rb

```diff
@@ -0,0 +1,120 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require 'md_to_pdf/core'
+
+module WorkPackage::PDFExport::Markdown
+  class MD2PDF
+    include MarkdownToPDF::Core
+
+    def initialize(styling_yml)
+      @styles = MarkdownToPDF::Styles.new(styling_yml)
+      init_options({ auto_generate_header_ids: false })
+      # @hyphens = Hyphen.new('en', false)
+    end
+
+    def draw_markdown(markdown, pdf, image_loader)
+      @pdf = pdf
+      @image_loader = image_loader
+      cm_extentions = %i[autolink strikethrough table tagfilter tasklist]
+      cm_parse_option = %i[FOOTNOTES SMART LIBERAL_HTML_TAG STRIKETHROUGH_DOUBLE_TILDE UNSAFE VALIDATE_UTF8]
+      root = CommonMarker.render_doc(markdown, cm_parse_option, cm_extentions)
+      begin
+        draw_node(root, pdf_root_options(@styles.page), true)
+      rescue StandardError => e
+        Rails.logger.error "Failed to draw markdown pdf: #{e}"
```

Comment:
Not really for development. My intention was to avoid that a PDF export with thousands of work packages would fail as whole if one `description` or `long text field` would somehow fail. E.g. when prawn can't fit a user generated table on a page at all. 
I should probably at least reduce the scope and catching `Prawn::Errors::CannotFit` only.

Reply:
Follow up: I did reduce the scope.

Reply:
If it's a background job, ideally it would report that to the job status so it becomes visible to the user in here: https://github.com/opf/openproject/blob/dev/app/workers/exports/export_job.rb#L118